ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5695705?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5695705?buildTab=artifacts#/) on master @ [54b4fcaa231abd16260fb09722ed291dbd6a4900](https://github.com/cockroachdb/cockroach/commits/54b4fcaa231abd16260fb09722ed291dbd6a4900):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint520174106
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5502: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (146.58s)
=== RUN   TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B
    helpers_test.go:702: making server as secondary tenant
    helpers_test.go:792: making cloudstorage feed factory
    changefeed_test.go:5494: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5494
        	            				helpers_test.go:856
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B
    --- FAIL: TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B (87.26s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-17470

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5701369?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5701369?buildTab=artifacts#/) on master @ [6374bd83f192bd509e50313acb431690b9c3f8ae](https://github.com/cockroachdb/cockroach/commits/6374bd83f192bd509e50313acb431690b9c3f8ae):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint2502934848
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5502: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (94.42s)
=== RUN   TestChangefeedBackfillCheckpoint/pubsub/limit=105_MB
    helpers_test.go:702: making server as secondary tenant
    helpers_test.go:792: making pubsub feed factory
    changefeed_test.go:5494: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5494
        	            				helpers_test.go:856
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/pubsub/limit=105_MB
    --- FAIL: TestChangefeedBackfillCheckpoint/pubsub/limit=105_MB (73.36s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5706737?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5706737?buildTab=artifacts#/) on master @ [86e007dcb5cbde3501f138c1d768519db3487857](https://github.com/cockroachdb/cockroach/commits/86e007dcb5cbde3501f138c1d768519db3487857):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint1686456074
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5502: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (92.50s)
=== RUN   TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB
    helpers_test.go:702: making server as secondary tenant
    helpers_test.go:792: making cloudstorage feed factory
    changefeed_test.go:5494: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5494
        	            				helpers_test.go:856
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB
    --- FAIL: TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB (79.35s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5731175?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5731175?buildTab=artifacts#/) on master @ [687171ac6c2cd9992486bb3b8c9d252ac95ca1cd](https://github.com/cockroachdb/cockroach/commits/687171ac6c2cd9992486bb3b8c9d252ac95ca1cd):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint610414776
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5503: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (146.60s)
=== RUN   TestChangefeedBackfillCheckpoint/webhook/limit=105_MB
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making webhook feed factory
    changefeed_test.go:5495: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5495
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/webhook/limit=105_MB
    --- FAIL: TestChangefeedBackfillCheckpoint/webhook/limit=105_MB (71.43s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5742095?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5742095?buildTab=artifacts#/) on master @ [88d3253301457ac57820e0f4a4fab8f74bf9f38b](https://github.com/cockroachdb/cockroach/commits/88d3253301457ac57820e0f4a4fab8f74bf9f38b):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint233736489
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5613: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (120.39s)
=== RUN   TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making cloudstorage feed factory
    changefeed_test.go:5605: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5605
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB
    --- FAIL: TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB (101.46s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5754067?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5754067?buildTab=artifacts#/) on master @ [e9ee21860458d997a8155734dc608cfcd050ef24](https://github.com/cockroachdb/cockroach/commits/e9ee21860458d997a8155734dc608cfcd050ef24):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint2482972554
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5613: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (165.00s)
=== RUN   TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making cloudstorage feed factory
    changefeed_test.go:5605: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5605
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB
    --- FAIL: TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB (85.12s)
=== RUN   TestChangefeedBackfillCheckpoint/webhook/limit=100_B
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making webhook feed factory
    changefeed_test.go:5605: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5605
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/webhook/limit=100_B
    --- FAIL: TestChangefeedBackfillCheckpoint/webhook/limit=100_B (79.71s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5765470?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5765470?buildTab=artifacts#/) on master @ [0e816efe5443764591a944edcb385c7b2762523e](https://github.com/cockroachdb/cockroach/commits/0e816efe5443764591a944edcb385c7b2762523e):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint2062998382
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5613: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (112.16s)
=== RUN   TestChangefeedBackfillCheckpoint/webhook/limit=100_B
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making webhook feed factory
    changefeed_test.go:5605: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5605
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/webhook/limit=100_B
    --- FAIL: TestChangefeedBackfillCheckpoint/webhook/limit=100_B (76.92s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5771318?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5771318?buildTab=artifacts#/) on master @ [e4cafeb8b1d586d091fb98e3e570650d7eeea294](https://github.com/cockroachdb/cockroach/commits/e4cafeb8b1d586d091fb98e3e570650d7eeea294):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint642691693
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5613: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (164.11s)
=== RUN   TestChangefeedBackfillCheckpoint/webhook/limit=105_MB
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making webhook feed factory
    changefeed_test.go:5605: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5605
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/webhook/limit=105_MB
    --- FAIL: TestChangefeedBackfillCheckpoint/webhook/limit=105_MB (75.58s)
=== RUN   TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making cloudstorage feed factory
    changefeed_test.go:5605: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5605
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B
    --- FAIL: TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B (88.38s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5776748?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5776748?buildTab=artifacts#/) on master @ [dc382b1ec54b4599cce1f5f50f243723a4286da6](https://github.com/cockroachdb/cockroach/commits/dc382b1ec54b4599cce1f5f50f243723a4286da6):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint1762800051
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    dir.go:42: unlinkat /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/TestChangefeedBackfillCheckpoint1592803917/0/2022-07-18: directory not empty
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5613: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (116.23s)
=== RUN   TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making cloudstorage feed factory
    changefeed_test.go:5605: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5605
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB
    --- FAIL: TestChangefeedBackfillCheckpoint/cloudstorage/limit=105_MB (83.97s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5789376?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5789376?buildTab=artifacts#/) on master @ [3dcf715f37595a73e8504b5331493f75207c6524](https://github.com/cockroachdb/cockroach/commits/3dcf715f37595a73e8504b5331493f75207c6524):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint3229525727
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5613: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (160.79s)
=== RUN   TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making cloudstorage feed factory
    changefeed_test.go:5605: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5605
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B
    --- FAIL: TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B (84.22s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5800939?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5800939?buildTab=artifacts#/) on master @ [b77d49e1aa199014c520a652624e801eb28893c5](https://github.com/cockroachdb/cockroach/commits/b77d49e1aa199014c520a652624e801eb28893c5):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/a77002d7c9453d7cd2d382f907780e13/logTestChangefeedBackfillCheckpoint1307303240
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestChangefeedBackfillCheckpoint
    changefeed_test.go:5613: -- test log scope end --
--- FAIL: TestChangefeedBackfillCheckpoint (170.89s)
=== RUN   TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B
    helpers_test.go:703: making server as secondary tenant
    helpers_test.go:793: making cloudstorage feed factory
    changefeed_test.go:5605: 
        	Error Trace:	helpers_test.go:180
        	            				helpers_test.go:248
        	            				changefeed_test.go:5605
        	            				helpers_test.go:857
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B
    --- FAIL: TestChangefeedBackfillCheckpoint/cloudstorage/limit=100_B (83.56s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82058 ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed [C-test-failure O-robot T-cdc branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/changefeedccl/cdctest/mock_webhook_sink.go
+++ b/pkg/ccl/changefeedccl/cdctest/mock_webhook_sink.go
@@ -29,6 +29,7 @@ type MockWebhookSink struct {
 		statusCodes      []int
 		statusCodesIndex int
 		rows             []string
+		notify           chan struct{}
 	}
 }
 
@@ -144,6 +145,19 @@ func (s *MockWebhookSink) Pop() string {
 	return ""
 }
 
+// NotifyMessage arranges for channel to be closed when message arrives.
+func (s *MockWebhookSink) NotifyMessage() chan struct{} {
+	c := make(chan struct{})
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if len(s.mu.rows) > 0 {
+		close(c)
+	} else {
+		s.mu.notify = c
+	}
+	return c
+}
+
 func (s *MockWebhookSink) requestHandler(hw http.ResponseWriter, hr *http.Request) {
 	method := hr.Method
 
@@ -177,7 +191,12 @@ func (s *MockWebhookSink) publish(hw http.ResponseWriter, hr *http.Request) erro
 	s.mu.numCalls++
 	if s.mu.statusCodes[s.mu.statusCodesIndex] >= http.StatusOK && s.mu.statusCodes[s.mu.statusCodesIndex] < http.StatusMultipleChoices {
 		s.mu.rows = append(s.mu.rows, string(row))
+		if s.mu.notify != nil {
+			close(s.mu.notify)
+			s.mu.notify = nil
+		}
 	}
+
 	hw.WriteHeader(s.mu.statusCodes[s.mu.statusCodesIndex])
 	s.mu.statusCodesIndex = (s.mu.statusCodesIndex + 1) % len(s.mu.statusCodes)
 	s.mu.Unlock()

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -5608,7 +5608,7 @@ func TestChangefeedBackfillCheckpoint(t *testing.T) {
 	// TODO(ssd): Tenant testing disabled because of use of DB()
 	for _, sz := range []int64{100 << 20, 100} {
 		maxCheckpointSize = sz
-		cdcTestNamedWithSystem(t, fmt.Sprintf("limit=%s", humanize.Bytes(uint64(sz))), testFn, feedTestEnterpriseSinks)
+		cdcTestNamedWithSystem(t, fmt.Sprintf("limit=%s", humanize.Bytes(uint64(sz))), testFn, feedTestForceSink("webhook"))
 	}
 }
 

--- a/pkg/ccl/changefeedccl/helpers_test.go
+++ b/pkg/ccl/changefeedccl/helpers_test.go
@@ -50,6 +50,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/log/eventpb"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
 	"github.com/lib/pq"
 	"github.com/stretchr/testify/require"
@@ -87,6 +88,7 @@ func readNextMessages(
 	ctx context.Context, f cdctest.TestFeed, numMessages int,
 ) ([]cdctest.TestFeedMessage, error) {
 	var actual []cdctest.TestFeedMessage
+	lastMessage := timeutil.Now()
 	for len(actual) < numMessages {
 		if ctx.Err() != nil {
 			return nil, ctx.Err()
@@ -94,11 +96,13 @@ func readNextMessages(
 		m, err := f.Next()
 		if log.V(1) {
 			if m != nil {
-				log.Infof(context.Background(), `msg %s: %s->%s (%s)`, m.Topic, m.Key, m.Value, m.Resolved)
+				log.Infof(context.Background(), `msg %s: %s->%s (%s) (%s)`,
+					m.Topic, m.Key, m.Value, m.Resolved, timeutil.Since(lastMessage))
 			} else {
 				log.Infof(context.Background(), `err %v`, err)
 			}
 		}
+		lastMessage = timeutil.Now()
 		if err != nil {
 			return nil, err
 		}
@@ -177,8 +181,14 @@ func assertPayloadsBase(
 	t testing.TB, f cdctest.TestFeed, expected []string, stripTs bool, perKeyOrdered bool,
 ) {
 	t.Helper()
+	timeout := assertPayloadsTimeout()
+	if len(expected) > 100 {
+		// Webhook sink is very slow; We have few tests that read 1000 messages.
+		timeout += 5 * time.Minute
+	}
+
 	require.NoError(t,
-		withTimeout(f, assertPayloadsTimeout(),
+		withTimeout(f, timeout,
 			func(ctx context.Context) error {
 				return assertPayloadsBaseErr(ctx, f, expected, stripTs, perKeyOrdered)
 			},

--- a/pkg/ccl/changefeedccl/sink_webhook.go
+++ b/pkg/ccl/changefeedccl/sink_webhook.go
@@ -40,7 +40,6 @@ const (
 	applicationTypeJSON = `application/json`
 	applicationTypeCSV  = `text/csv`
 	authorizationHeader = `Authorization`
-	defaultConnTimeout  = 3 * time.Second
 )
 
 func isWebhookSink(u *url.URL) bool {
@@ -303,10 +302,9 @@ func makeWebhookSink(
 		return nil, errors.Errorf(`this sink requires the WITH %s option`, changefeedbase.OptTopicInValue)
 	}
 
-	connTimeout := opts.ClientTimeout
-	if connTimeout == nil {
-		t := defaultConnTimeout
-		connTimeout = &t
+	var connTimeout time.Duration
+	if opts.ClientTimeout != nil {
+		connTimeout = *opts.ClientTimeout
 	}
 
 	ctx, cancel := context.WithCancel(ctx)
@@ -328,7 +326,7 @@ func makeWebhookSink(
 	}
 
 	// TODO(yevgeniy): Establish HTTP connection in Dial().
-	sink.client, err = makeWebhookClient(u, *connTimeout)
+	sink.client, err = makeWebhookClient(u, connTimeout)
 	if err != nil {
 		return nil, err
 	}

--- a/pkg/ccl/changefeedccl/testfeed_test.go
+++ b/pkg/ccl/changefeedccl/testfeed_test.go
@@ -1429,7 +1429,9 @@ func (f *webhookFeedFactory) Feed(create string, args ...interface{}) (cdctest.T
 
 		if createStmt.SinkURI == nil {
 			createStmt.SinkURI = tree.NewStrVal(
-				fmt.Sprintf("webhook-%s?insecure_tls_skip_verify=true&client_cert=%s&client_key=%s", sinkDest.URL(), base64.StdEncoding.EncodeToString(clientCertPEM), base64.StdEncoding.EncodeToString(clientKeyPEM)))
+				fmt.Sprintf("webhook-%s?insecure_tls_skip_verify=true&client_cert=%s&client_key=%s",
+					sinkDest.URL(), base64.StdEncoding.EncodeToString(clientCertPEM),
+					base64.StdEncoding.EncodeToString(clientKeyPEM)))
 		}
 	} else {
 		sinkDest, err = cdctest.StartMockWebhookSink(cert)
@@ -1582,6 +1584,7 @@ func (f *webhookFeed) Next() (*cdctest.TestFeedMessage, error) {
 		case <-time.After(timeout()):
 			return nil, &contextutil.TimeoutError{}
 		case <-f.ss.eventReady():
+		case <-f.mockSink.NotifyMessage():
 		case <-f.shutdown:
 			return nil, f.terminalJobError()
 		}

